java/src/jmri/jmrit/symbolicprog/tabbedframe/PaneProgFrame.java

```diff
@@ -1525,22 +1479,27 @@ public void prepGlassPane(AbstractButton activeButton) {
         if (activeButton == readChangesButton || activeButton == readAllButton
                 || activeButton == writeChangesButton || activeButton == writeAllButton) {
             if (activeButton == readChangesButton) {
-                for (int i = 0; i < paneList.size(); i++) {
-                    activeComponents.add(((PaneProgPane) paneList.get(i)).readChangesButton);
+                for (JPanel jPanel : paneList) {
+                    Assume.assumeTrue(jPanel instanceof PaneProgPane);
```

Comment:
I think you can replace these with:
`assert jPanel instanceof PaneProgPane;`

See:
https://stackoverflow.com/questions/2758224/what-does-the-java-assert-keyword-do-and-when-should-it-be-used
https://www.baeldung.com/java-assert

Reply:
Thanks, I was puzzled as to why it failed